pkg/bgpv1/agent/controller_test.go

```diff
@@ -139,23 +137,24 @@ func TestControllerSanity(t *testing.T) {
 				return []*v2alpha1api.CiliumBGPPeeringPolicy{p}, nil
 			},
 			configurePeers: func(_ context.Context, p *v2alpha1api.CiliumBGPPeeringPolicy, c *agent.ControlPlaneState) error {
-				defaulted := false
 				for _, r := range p.Spec.VirtualRouters {
 					for _, n := range r.Neighbors {
-						if n.PeerPort != nil &&
-							*n.PeerPort == types.DefaultPeerPort &&
-							n.ConnectRetryTime.Duration != 0 &&
-							n.HoldTime.Duration != 0 &&
-							n.KeepAliveTime.Duration != 0 &&
-							n.GracefulRestart.RestartTime.Duration != 0 {
-
-							defaulted = true
+						if n.PeerPort == nil ||
+							n.ConnectRetryTimeSeconds == nil ||
+							n.HoldTimeSeconds == nil ||
+							n.KeepAliveTimeSeconds == nil ||
+							n.GracefulRestart.RestartTimeSeconds == nil {
+							t.Fatalf("policy: %v not defaulted properly (nil)", p)
+						}
+						if *n.PeerPort != v2alpha1api.DefaultBGPPeerPort ||
+							*n.ConnectRetryTimeSeconds != v2alpha1api.DefaultBGPConnectRetryTimeSeconds ||
+							*n.HoldTimeSeconds != v2alpha1api.DefaultBGPHoldTimeSeconds ||
+							*n.KeepAliveTimeSeconds != v2alpha1api.DefaultBGPKeepAliveTimeSeconds ||
+							*n.GracefulRestart.RestartTimeSeconds != v2alpha1api.DefaultBGPGRRestartTimeSeconds {
+							t.Fatalf("policy: %v not defaulted properly (invalid value)", p)
```

Comment:
Hmm, I'm not sure about that - this is a unit test, something very internal. The API package is on the other hand something very external, and such a function would not have any other practical use than this test. But at the same time I understand the inconvenience.. Would like to know what the others think about it :)

Reply:
Two comments regarding this 
1. Do you think this test should move into bgpp_types_test.go? Since, we moved SetDefaults() method there, shouldn't the test to validate that Defaults are set correctly be moved within that pkg? This should make it easier to maintain, as both code and tests live in the same pkg ( harder to miss ).
2. I tend to agree with @rastislavs  on not adding a new method externally for something which is used solely in the test.

Reply:
I see, why not split the test into two parts? 

- Test_SetDefaults (in bgpp_types_test.go): Validates if values are correctly set to expected defaults 
- Test_Policy (in controller_test.go): Validates there are no nil values. 

No strong opinion, just bouncing ideas. :)

Reply:
In https://github.com/cilium/cilium/pull/26040/commits/bc81751823d7d638426726f47f3ca861c786a081 I added unit tests to the API package and removed checks for default values in the controller unit test.